package.json

```diff
@@ -29,15 +29,15 @@
     "url": "https://github.com/Pomax/react-onclickoutside/issues"
   },
   "scripts": {
-    "build": "npm run build:es && npm run build:commonjs",
+    "build": "run-p build:*",
     "build:es": "cross-env BABEL_ENV=es babel src -d es",
     "build:commonjs": "cross-env BABEL_ENV=commonjs babel src -d lib",
     "lint": "eslint src/*.js ./test",
     "test": "run-s test:**",
-    "test:basic": "run-s lint build",
+    "test:basic": "run-s lint build:commonjs",
     "test:karma": "karma start test/karma.conf.js --single-run",
     "test:nodom": "mocha test/no-dom-test.js",
-    "precommit": "npm test && lint-staged",
+    "precommit": "lint-staged && npm test",
```

Comment:
hm, u are quite right, its a duplication (as it already was :D)

now its `lint-staged -> lint (all) -> build:commonjs -> run tests`
before it was - `lint (all) -> build:commonjs -> run tests -> lint-staged`

I've changed the order as I wanted to format the changed files first and THEN run tests as reformatting code might lead (in extreme situation, this should never happen!) to some unexpected changes. So it seemed reasonable to run tests after they get reformatted.

Personally i would remove linting from `test` task, those are imho 2 different things anyway. Linting errors still will be catched before commiting thanks to `lint-staged`